src/util/fields.py

```diff
@@ -1008,6 +1008,34 @@ def adapt_dict(d):
             )
 
 
+# adapt depends for custom compute fields
+def adapt_depends(cr, model, old, new):
```

Comment:
- missing loop on inherits
 - verify the model

```suggestion
def adapt_depends(cr, model, old, new, skip_inherit=()):
    _validate_model(model)
```